src/app/cli/commands/configure.ts

```diff
@@ -0,0 +1,856 @@
+import * as p from '@clack/prompts';
+import chalk from 'chalk';
+import fs from 'fs/promises';
+import path from 'path';
+import * as YAML from 'yaml';
+import {
+    MCP_SERVER_REGISTRY,
+    getMcpServerCategories,
+    getMcpServersByCategory,
+} from '@core/config/mcp-registry.js';
+import { ConfigurationManager } from '@core/config/config-manager.js';
+import {
+    getDefaultModelForProvider,
+    LLMProvider,
+    logger,
+    getSupportedModels,
+} from '@core/index.js';
+import type { AgentConfig, McpServerConfig } from '@core/config/schemas.js';
+import { getPrimaryApiKeyEnvVar } from '@core/utils/api-key-resolver.js';
+import os from 'os';
+
+export interface ConfigureCommandOptions {
+    /** Whether to save the configuration after creation */
+    save?: boolean;
+    /** Output path for the configuration file */
+    output?: string;
+    /** Load an existing configuration to modify */
+    load?: string | boolean;
+    /** List saved configurations */
+    list?: boolean;
+    /** Delete a saved configuration */
+    delete?: string;
+    /** Export a saved configuration */
+    export?: string;
+    /** Quick path (minimal prompts) */
+    quick?: boolean;
+}
+
+// ────────────────────────── Local MCP registry helpers ─────────────────────────
+const LOCAL_MCP_REGISTRY_PATH = path.join(os.homedir(), '.saiki', 'mcp-registry.local.json');
+
+async function loadLocalMcpRegistry(): Promise<Record<string, any>> {
+    try {
+        const data = await fs.readFile(LOCAL_MCP_REGISTRY_PATH, 'utf-8');
+        return JSON.parse(data);
+    } catch (_err) {
+        return {};
+    }
+}
+
+/**
+ * Main configure command - provides interactive agent configuration
+ */
+export async function configureCommand(options: ConfigureCommandOptions = {}): Promise<void> {
+    const configManager = new ConfigurationManager();
+
+    try {
+        // Handle list command
+        if (options.list) {
+            await listConfigurations(configManager);
+            return;
+        }
+
+        // Handle delete command
+        if (options.delete) {
+            await deleteConfiguration(configManager, options.delete);
+            return;
+        }
+
+        // Handle export command
+        if (options.export) {
+            await exportConfiguration(configManager, options.export, options.output);
+            return;
+        }
```

Comment:
these might be better served as separate sub-commands instead of arguments

`saiki config create` - interactive CLI for creating configs
`saiki config list` - lists all configs
`saiik config export` - exports the config
`saiki config delete` - deletes the config

this is more in line with existing CLIs which usually chain subcommands like this
leads to better flow where each sub-command can evolve independently and have different arguments

`saiki -h` -> mentions saiki config as a sub-command for managing/creating configs
`saiki config -h` -> lists all the sub-commands for managing config
`saiki config create -h` -> how to use the create sub-command
`saiki config delete -h` -> how to use the delete sub-command